lib/_tls_common.js

```diff
@@ -195,6 +195,13 @@ exports.createSecureContext = function createSecureContext(options, context) {
     c.context.setFreeListLength(0);
   }
 
+  if (typeof options.clientCertEngine === 'string') {
+    if (c.context.setClientCertEngine)
+      c.context.setClientCertEngine(options.clientCertEngine);
+    else
+      throw new Error('Custom engines not supported by this OpenSSL');
+  }
```

Comment:
`null` and `undefined` should be allowed (and ignored), but perhaps any other type should be rejected as invalid usage?

Reply:
Sure, done. (Still have to add a test, but since this has to be moved to the new error system anyway, I've made a note-to-self to add a test then.)